### PR TITLE
Feature/sc 246044/migrate at core databricks modules

### PR DESCRIPTION
**Pull request by aarroyosal**
shortcut tickets:
- https://app.shortcut.com/cartoteam/story/246038/add-at-databricks-scaffolding
- https://app.shortcut.com/cartoteam/story/246044/migrate-at-core-databricks-modules

Acceptance:

Install the libraries needed for the databricks AT (README)
[Setup databricks databricks CLI authentication (via token)](https://docs.databricks.com/dev-tools/cli/index.html#set-up-authentication)
test the make commands
make build|clean|lint|deploy CLOUD=databricks
test the deployed functions in a databricks notebook `select carto_dev_data.st_x(geom) as latitude from...`

### CODE REVIEW

File: clouds/databricks/common/Makefile [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	dbfs cp --overwrite $(JAR_DIR)/core-assembly-*-SNAPSHOT.jar $(JAR_DEPLOY_PATH)/analyticstoolbox-assembly-SNAPSHOT.jar
+	echo "Jars uploaded"
+
+upload-sql:
+ifndef DB_SCHEMA
+	echo "Databricks schema not defined, using default"
+	$(eval DB_SCHEMA := default)
+	echo $(DB_SCHEMA)
+endif
+	sed -i '.bkp' '1s/^/USE ${DB_SCHEMA};\n/' $(SQL_PATH)
```
Comment (Jesus89) [inline clouds/databricks/common/Makefile:67]: To make this work in GNU/Linux must be `sed -i'.bkp'` (without the space)

**Review by Jesus89 — CHANGES_REQUESTED** [review]
Awesome job!

I left some in-line comments. I would remove the entire cloud/databricks/.github section for now. In general for the docs, we should review the style of the examples to follow the convention:
- Uppercase syntax: SELECT, AS, etc
- Ending with ;

NOTE: we leave the automatic generation of createUDFs to the next PR.

File: clouds/databricks/.github/release-drafter.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name-template: '$NEXT_MINOR_VERSION'
```
[the remark this answers is not in the document]
Comment (aarroyosal) [reply clouds/databricks/.github/release-drafter.yml]: Done

File: clouds/databricks/README.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
+# CARTO Analytics Toolbox
```
[the remark this answers is not in the document]
Comment (aarroyosal) [reply clouds/databricks/README.md:1]: I have removed the badges and updated the links. I have read the rest and looks good to me

File: clouds/databricks/common/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,16 @@
+attrs==22.1.0
```
[the remark this answers is not in the document]
Comment (aarroyosal) [reply clouds/databricks/common/requirements.txt]: Yeah, they are dependencies of dependencies, that's how pip freeze works. I have already cleaned them

File: clouds/databricks/libraries/scala/LICENSE [hunk 5]
```diff
@@ -0,0 +1,14 @@
+SPDX short identifier: BSD-3-Clause
+
+Copyright (c) 2021, CARTO
```
[the remark this answers is not in the document]
Comment (aarroyosal) [reply clouds/databricks/libraries/scala/LICENSE]: Changed

File: clouds/databricks/modules/accessors/doc/ST_COORDDIM.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+* `geom`: `Geometry` input geom.
+
+**Return type**
+
+`Int`
+
+**Example**
+
+``` sql
+SELECT carto.ST_COORDDIM(ST_MAKEPOINTM(1, 2, 3));;
```
[the remark this answers is not in the document]
Comment (aarroyosal) [reply clouds/databricks/modules/accessors/doc/ST_COORDDIM.md]: Done

File: clouds/databricks/modules/accessors/doc/ST_DIMENSION.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+* `geom`: `Geometry` input geom.
+
+**Return type**
+
+`Int`
+
+**Example**
+
+``` sql
+SELECT ST_DIMENSION(ST_GEOMFROMWKT("LINESTRING(0 0, 1 1)"))
```
[the remark this answers is not in the document]
Comment (aarroyosal) [reply clouds/databricks/modules/accessors/doc/ST_DIMENSION.md]: Done

File: clouds/databricks/modules/predicates/doc/ST_CROSSES.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+**Return type**
+
+`Boolean`
+
+**Example**
+
+```sql
+WITH t AS (
+  select ST_GEOMFROMWKT("LINESTRING (1 0, 1 2)") as lineA,
```
[the remark this answers is not in the document]
Comment (aarroyosal) [reply clouds/databricks/modules/predicates/doc/ST_CROSSES.md]: Changed

File: clouds/databricks/libraries/scala/core/src/main/scala/com/carto/analyticstoolbox/index/ST_ExtentFromGeom.scala [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+package com.carto.analyticstoolbox.index
+
+import com.azavea.hiveless.HUDF
+import com.carto.analyticstoolbox.modules._
+import geotrellis.vector._
+
+class ST_ExtentFromGeom extends HUDF[Geometry, Extent] {
+  def function = _.extent
```
[the remark this answers is not in the document]
Comment (aarroyosal) [reply clouds/databricks/libraries/scala/core/src/main/scala/com/carto/analyticstoolbox/index/ST_ExtentFromGeom.scala]: Fixed

**Review by Jesus89 — APPROVED** [review]
LGTM

**Pull request merged**